shopfloor/services/service.py

```diff
@@ -46,21 +63,41 @@ def search_move_line(self):
         # every time.
         return self._actions_for("search_move_line", picking_types=self.picking_types)
 
-    def _check_picking_status(self, pickings, states=("assigned",)):
-        """Check if given pickings can be processed.
+    def _check_picking_consistency(self, pickings):
+        if not pickings.exists():
+            return self.msg_store.stock_picking_not_found()
+
+    def _check_picking_type(self, pickings):
+        """Check if the pickings have the right expected type."""
+        if not any(
+            picking.picking_type_id in self.picking_types for picking in pickings
+        ):
+            return self.msg_store.reserved_for_other_picking_type(
+                fields.first(pickings)
+            )
 
-        If the picking is already done, canceled or didn't belong to the
-        expected picking type, a message is returned.
-        """
-        for picking in pickings:
-            if not picking.exists():
-                return self.msg_store.stock_picking_not_found()
-            if picking.state == "done":
-                return self.msg_store.already_done()
-            if picking.state not in states:  # the picking must be ready
-                return self.msg_store.stock_picking_not_available(picking)
-            if picking.picking_type_id not in self.picking_types:
-                return self.msg_store.cannot_move_something_in_picking_type()
+    def _check_picking_status(self, pickings, states=("assigned",)):
+        """Checks if the picking exists, is already done or canceled."""
+        if not any(picking.state != "done" for picking in pickings):
+            return self.msg_store.already_done()
+        if not any(picking.state != "cancel" for picking in pickings):
+            return self.msg_store.transfer_canceled()
+        if not any(
+            picking.state in states for picking in pickings
+        ):  # the picking must be ready
+            return self.msg_store.stock_picking_not_available(fields.first(pickings))
```

Comment:
Why was this changed to `not any`?
For the done check, before it returned the message if any picking is done, now it returns the messages only if all pickings are done.